apps/web/src/services/cost-explorer/composables/queries/use-cost-query-set-query.ts

```diff
@@ -1,22 +1,61 @@
-import { computed } from 'vue';
+import { computed, type Ref, type ComputedRef } from 'vue';
 
 import type { ConsoleFilter } from '@cloudforet/core-lib/query/type';
 import { ApiQueryHelper } from '@cloudforet/core-lib/space-connector/helper';
 
 import { useCostQuerySetApi } from '@/api-clients/cost-analysis/cost-query-set/composables/use-cost-query-set-api';
+import type { CostQuerySetModel } from '@/api-clients/cost-analysis/cost-query-set/schema/model';
 import { useScopedQuery } from '@/query/composables/use-scoped-query';
 import { useServiceQueryKey } from '@/query/query-key/use-service-query-key';
 
 import { useAppContextStore } from '@/store/app-context/app-context-store';
 import { useUserStore } from '@/store/user/user-store';
 
-export const useCostQuerySetQuery = (data_source_id: string) => {
+import { UNIFIED_COST_KEY } from '@/services/cost-explorer/constants/cost-explorer-constant';
+import { ADMIN_MANAGED_COST_QUERY_SET_LIST, MANAGED_COST_QUERY_SET_IDS, MANAGED_COST_QUERY_SET_LIST } from '@/services/cost-explorer/constants/managed-cost-analysis-query-sets';
+
+interface UseCostQuerySetQueryOptions {
+    data_source_id: string | Ref<string> | ComputedRef<string>;
+    isUnifiedCostOn?: boolean | Ref<boolean> | ComputedRef<boolean>;
+    selectedQuerySetId?: string | Ref<string> | ComputedRef<string> | Ref<string | undefined> | ComputedRef<string | undefined>;
+}
+
+export const useCostQuerySetQuery = (options: UseCostQuerySetQueryOptions) => {
     const appContextStore = useAppContextStore();
     const userStore = useUserStore();
     const _userId = computed<string>(() => userStore.state.userId ?? '');
     const _isAdminMode = computed<boolean>(() => appContextStore.getters.isAdminMode);
     const _workspaceId = computed<string|undefined>(() => appContextStore.getters.workspaceId);
 
+    // Convert options to computed refs for reactivity
+    const _dataSourceId = computed(() => {
+        const value = typeof options.data_source_id === 'string' ? options.data_source_id : options.data_source_id.value;
+        return value;
+    });
+    const _isUnifiedCostOn = computed(() => {
+        const value = options.isUnifiedCostOn;
+        return typeof value === 'boolean' ? value : value?.value ?? false;
+    });
+    const _selectedQuerySetId = computed(() => {
+        const value = options.selectedQuerySetId;
+        return typeof value === 'string' ? value : value?.value;
+    });
```

Comment:
information: If you use `string | Ref<string> | ComputedRef<string>`, you can use `toValue` in `vueuse` for check value!!